[DRAFT] Add docker image tests demo

ZhouXing19 opened the pull request:
This is a draft PR to test the teamcity CI for docker image.
It should never be merged and will be closed once the tests are done.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/75278)
<!-- Reviewable:end -->

ZhouXing19 committed:
bazel: run docker image tests under Bazel

This commit is to add a test infrastructure for docker image in TeamCity using
bazel.
It downloads the docker image tar from an upstream build configuration
(Build Docker Image), loads the docker image from the tar,
builds a docker container based on it, and runs SQL queries inside that
container.

Release note: None

ZhouXing19 committed:
docker: increase the memory allocated to the `cockroach-ci` image in CI.

We encountered an exit code of `137` when running docker image test in the CI,
and this means that the container has run out of allocated memory.
This commit is to increase the allocated memory to 30GB, and set unlimited
memory swap to disk.
Release note: None
